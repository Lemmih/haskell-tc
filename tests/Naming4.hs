module Naming4 where

{-
fn var_a = sub var_a
  where
    sub another_a = const var_a another_a
-}
const _ b = b

{-
outer a = a
  where
    inner b = b
-}

outer a = a
  where
    inner b = const (const a ()) b
    id x = x
