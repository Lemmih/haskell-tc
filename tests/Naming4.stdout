Bindings:
:: âˆ€ a b. a â†’  b â†’  b
  [4mconst _ b = b[0m
:: b
  const _ [4mb[0m = b
:: âˆ€ a. a â†’  a
  [4mouter a = a[0m
  [4m  where[0m
  [4m    inner b = const (const a ()) b[0m
  [4m    id x = x[0m
:: a
  outer [4ma[0m = a
:: âˆ€ b. b â†’  b
      [4minner b = const (const a ()) b[0m
:: b
      inner [4mb[0m = const (const a ()) b
:: âˆ€ c. c â†’  c
      [4mid x = x[0m
:: c
      id [4mx[0m = x

Proofs:
coercion: b
  const _ b = [4mb[0m
coercion: a
  outer a = [4ma[0m
coercion: (âˆ€ a b. a â†’  b â†’  b) @ () b
      inner b = [4mconst[0m (const a ()) b
coercion: (âˆ€ a b. a â†’  b â†’  b) @ a ()
      inner b = const ([4mconst[0m a ()) b
coercion: a
      inner b = const (const [4ma[0m ()) b
coercion: b
      inner b = const (const a ()) [4mb[0m
coercion: c
      id x = [4mx[0m
